postreise/plot/plot_capacity_map.py

```diff
@@ -0,0 +1,190 @@
+    else:
+        all_plot_states_kwargs = default_plot_states_kwargs
+    plot_states(bokeh_figure=p, **all_plot_states_kwargs)
+
+    add_plant_capacity(
+        p,
+        scenario,
+        resources,
+        disaggregation,
+        min_capacity,
+        size_factor,
+        alpha,
+    )
+
+    p.legend.label_text_font_size = f"{legend_font_size}pt"
+    p.legend.location = legend_location
+
+    return p
+
+
+def add_plant_capacity(
+    bokeh_figure,
+    scenario,
+    resources,
+    disaggregation=None,
+    min_capacity=1,
+    size_factor=1,
+    alpha=0.5,
+):
+    """Adds renewables capacity to a plot.
+
+    :param bokeh.plotting.figure.Figure bokeh_figure: figure to plot capacities onto.
+    :param powersimdata.scenario.scenario.Scenario scenario: scenario instance.
+    :param iterable resources: which types of resources to plot.
+    :param tuple figsize: size of the bokeh figure (in pixels).
+    :param tuple(float, float) x_range: x range to zoom plot to (EPSG:3857).
+    :param tuple(float, float) y_range: y range to zoom plot to (EPSG:3857).
+    :param str disaggregation: method used to disaggregate plants:
+        if "new_vs_existing_plants": separates plants into added vs. existing.
+        if None, no disaggregation.
+    :param dict plot_states_kwargs: keyword arguments to pass to
+        :func:`postreise.plot.plot_states.plot_states`.
+    :param float/int min_capacity: minimum bus capacity (MW) for markers to be plotted.
+    :param float/int size_factor: scale size of glyphs.
+    :param float/int alpha: opacity of circles (between 0 and 1).
+    """
+    ct = scenario.get_ct()
+    grid = scenario.get_grid()
+    grid.plant["lat"] = grid.plant["lat"].round(3)
+    grid.plant["lon"] = grid.plant["lon"].round(3)
+    grid.plant = project_bus(grid.plant.query("type in @resources"))
+    type_colors = grid.model_immutables.plants["type2color"]
+
+    xy_capacity = {}
+    if disaggregation is None:
+        grouped_capacities = grid.plant.groupby(["x", "y", "type"]).sum().Pmax
+        xy_capacity["all"] = grouped_capacities.loc[grouped_capacities > 0]
+    elif disaggregation == "new_vs_existing_plants":
+        if "new_plant" in ct.keys():
+            num_new_plants = len(ct["new_plant"])
+            scaled_plants = grid.plant.iloc[:-num_new_plants]
+            new_plants = grid.plant.iloc[-num_new_plants:]
+            grouped_new_capacities = new_plants.groupby(["x", "y", "type"]).sum().Pmax
+            xy_capacity["new"] = grouped_new_capacities.loc[grouped_new_capacities > 0]
+        else:
+            scaled_plants = grid.plant
+            new_plants = pd.DataFrame(columns=grid.plant.columns)
+            xy_capacity["new"] = pd.Series(
+                index=pd.MultiIndex(
+                    levels=[[], [], []],
+                    codes=[[], [], []],
+                    names=("x", "y", "type"),
+                ),
+            )
```

Comment:
Is there a specific reason that we would like to maintain the `multiIndex` structure instead of simple do a `reset_index()` after `groupby.sum`? This might also simplify the data accessing below when generating the capacity circles.

Reply:
I had no particular reason to maintain the MultiIndex. If you think it would look cleaner to reset it and be able to use more standard `.loc` slicing, I can change it.

Reply:
Yeah, it will be much easier to mimic a `dataframe` than a `multiIndex` `series` here and we could get rid of `get_level_values()` below to make it more readable. That's my personal preference though, it is fine to keep it as it is, too.

Reply:
Done. Tested successfully with and without disaggregation using:
```python
from bokeh.io import show
from powersimdata import Scenario
from postreise.plot.plot_capacity_map import map_plant_capacity

resources = {"solar", "wind", "nuclear"}
plot_kwargs = {
    "x_range": (-1.4e7, -1.12e7),
    "y_range": (3.6e6, 6.4e6),
    "figsize": (800, 800),
}

show(map_plant_capacity(Scenario(3697), resources, **plot_kwargs))
show(map_plant_capacity(Scenario(3697), resources, disaggregation="new_vs_existing_plants", **plot_kwargs))
```